### ISSUE TITLE
High CPU usage when paused

### ISSUE THREAD

**Issue by itsnotoger** [caprica/vlcj#492]
I found an issue that occurs when pausing the player that is playing a live stream. After a lot of testing (with my implementation using a DirectMediaPlayer for JavaFX (ResizableJavaFXPlayer from the vlcj-javafx repo), the TestPlayer from this repo and the regular VLC Player) it seems to be related to the download of the stream segments. The overall CPU usage for the VLCJ Java process on my PC (i5-3570K CPU @ 3.40GHz) goes to about 30% (+/- 2%) when the player is paused (a pause prevents VLC player from downloading more segments).
Attached should be an image to help explain the problem.
![image](https://cloud.githubusercontent.com/assets/19309683/15187952/9e5ce43c-17a3-11e6-982d-aa9a8087ccd4.png)
The URL for the type of stream I used looks something like this [http://video36.fra01.hls.ttvnw.net/hls-80ed04/nightblue3_21261840576_451011923/high/index-live.m3u8?token=id=49823466292385828,bid=21261840576,exp=1463067528,node=video36-1.fra01.hls.justin.tv,nname=video36.fra01,fmt=high&sig=4b241bd2b06a78534357b35700d45d63a689451a](url)
The main problem I have is that the CPU usage is very high at a time where I expect almost no usage due to nothing being played or loaded. I could understand that during playback the CPU usage keeps going up and down (although even then I would consider it a high CPU drain) but when the player is paused this is not something I expect. The CPU usage is also very irregular even when the stream being played is smooth.
As I said I have observed this for a while and the correlation between CPU and Ethernet is as follows: A chunk of the stream is being downloaded and and soon as this finishes the CPU usage goes up until exactly the next chunk is being downloaded, where it goes down again. This might not sound like a problem but pausing the player at certain times show something more. In the image I added grey bars to the end of a chunk of the stream that finished downloading. The CPU usage increases as described from there on until the next download begins. The black bar is when I paused the player. After this pause the CPU usage will stay at this level forever if the player is left alone. Interestingly when I pause the player **during** the download of a chunk, the CPU usage will go down to almost nothing (as expected).
I am guessing this behavior is the same for all VLCJ implementations while this is not an issue with VLC Player where the CPU usage does not increase even when paused at a time where no chunk download is occuring. I have not tested any other type of stream and the given URL might not even be valid for anyone else. If necessary I can provide instructions on how to obtain this type of stream URL (twitch.tv live stream). My max download speed is 3 Mbps in case that matters.

**Comment by caprica** [comment]
Are you saying the normal "embedded" media player shows the same symptoms?

**Comment by itsnotoger** [comment]
I tested this with src/test/java/uk/co/caprica/vlcj/test/basic/TestPlayer.java which I assume is the embedded media player. The symptoms are the same.

**Comment by caprica** [comment]
If that is the case, then it's doubtful there is anything to fix from vlcj's point of view.

With the "embedded" media player, vlcj really does nothing other than tell VLC to render the video in a particular heavyweight component (an AWT Canvas). This is in contrast to the "direct" rendering in the JavaFX player where it is vlcj and your own code that renders the video.

Right now I don't know what else to say.

**Comment by itsnotoger** [comment]
I could imagine some downloader thread busy waiting until it receives a new job in VLC but I was not sure why this is the case using VLCJ and not VLC Player. I have not checked for differences in the libvlc versions, so that could be the problem. Either way this can be closed if VLCJ does not even touch anything regarding downloads.

**Comment by caprica** [comment]
vlcj definitely does not do anything with regards to downloading the stream data.

The question about the different behaviour between vlcj and VLC itself is why I asked about the "embedded" vs "direct" media player. In theory, the embedded version should behave the same as VLC as much as possible.

**Comment by itsnotoger** [comment]
Perhaps there are some parameters that libvlc takes which influence the behavior of downloads?

**Comment by caprica** [comment]
As far as I know, there are only such options for caching.

You may check yourself by typing "vlc -H" at a command-line.

**Comment by itsnotoger** [comment]
I think I just solved the problem. I checked for version differences and I was using VLC version 2.2.1 64bit in Java and the regular VLC Player I was testing with was 2.2.2 32bit. The reason for the version differences was that the 64bit versions of VLC Player kept crashing for me, although not libvlc itself used with VLCJ. I replaced it with 2.2.3 64bit just now and the CPU usage seems as expected. 
Apologies for 'blaming' VLCJ with this issue :(
